packages/shared/src/components/profile/UserShortInfo.tsx

```diff
@@ -30,7 +31,9 @@ export interface UserInfoClassName {
 export interface UserShortInfoProps<
   Tag extends React.ElementType = React.ElementType,
 > {
-  user: UserShortProfile;
+  user: UserShortProfile & {
+    role?: SourceMemberRole;
```

Comment:
Why don't we just add the `role` as a separate prop? To avoid polymorphing our common User entity and make it easier to use elsewhere.